extensions/terminal-suggest/src/tokens.ts

```diff
@@ -14,20 +14,102 @@ export const enum TokenType {
 const shellTypeResetChars = new Map<TerminalShellType, string[]>([
 	[TerminalShellType.Bash, ['>', '>>', '<', '2>', '2>>', '&>', '&>>', '|', '|&', '&&', '||', '&', ';', '(', '{', '<<']],
 	[TerminalShellType.Zsh, ['>', '>>', '<', '2>', '2>>', '&>', '&>>', '<>', '|', '|&', '&&', '||', '&', ';', '(', '{', '<<', '<<<', '<(']],
-	[TerminalShellType.PowerShell, ['>', '>>', '<', '2>', '2>>', '*>', '*>>', '|', '-and', '-or', '-not', '!', '&', '-eq', '-ne', '-gt', '-lt', '-ge', '-le', '-like', '-notlike', '-match', '-notmatch', '-contains', '-notcontains', '-in', '-notin']]
+	[TerminalShellType.PowerShell, ['>', '>>', '<', '2>', '2>>', '*>', '*>>', '|', '-and', '-or', '-not', '!', '&', ';', '-eq', '-ne', '-gt', '-lt', '-ge', '-le', '-like', '-notlike', '-match', '-notmatch', '-contains', '-notcontains', '-in', '-notin']]
+]);
+
+// Command separators that start new command contexts (vs logical operators that stay in argument context)
+const shellTypeCommandSeparators = new Map<TerminalShellType, string[]>([
+	[TerminalShellType.Bash, ['|', '|&', '&&', '||', '&', ';', '(', '{']],
+	[TerminalShellType.Zsh, ['|', '|&', '&&', '||', '&', ';', '(', '{']],
+	[TerminalShellType.PowerShell, ['|', '&', ';']]
 ]);
 
 const defaultShellTypeResetChars = shellTypeResetChars.get(TerminalShellType.Bash)!;
+const defaultShellTypeCommandSeparators = shellTypeCommandSeparators.get(TerminalShellType.Bash)!;
+
+export { shellTypeResetChars, defaultShellTypeResetChars, shellTypeCommandSeparators, defaultShellTypeCommandSeparators };
 
 export function getTokenType(ctx: { commandLine: string; cursorPosition: number }, shellType: TerminalShellType | undefined): TokenType {
```

Comment:
for this function, add examples to comments so it's easier to understand/ read each case